pytorch_blade/pytorch_blade/compiler/mlir/runtime/ral_context.cpp

```diff
@@ -263,6 +269,25 @@ tao::ral::BaseContext* RalContext::LoadCache() {
   }
   return ral_ctx_ptr;
 }
+
+// Because weight is loaded by RAL lazily on the first inference. And there's no
+// way to to move loaded weight to another devices in RAL currently. So we need
+// to make sure no change on current device during inference. So the reasonable
+// restriction is to deny device change during inferences but allow it before
+// the first inference.
+int64_t RalContext::CheckAndGetGPUDevice() {
+  int64_t cur_device = c10::cuda::current_device();
+  int64_t prev_device = NULL_GPU_DEVICE;
```

Comment:
prev_device -> null_device ?

Reply:
after compare_exchange on `this->gpu_device_`, pre_device will be filled with the truly previous device. It's just initialized with a 'null' value.